From e7ea8e7f5e58d5de1f52e4ae48c683d20bb823e6 Mon Sep 17 00:00:00 2001
From: Peng Ma <peng.ma@nxp.com>
Date: Wed, 25 Jul 2018 08:53:07 +0000
Subject: [PATCH 593/767] dts: fsl-ls1021a, fsl-ls1043a, fsl-ls1046a: add multi
 block node support

add block-offset to support different virtual block offset for qdma
base on soc;
the interrupt named "qdma-queueN(N:0,1,2,3)" correspond to a virtual
block,N based on block number of qdma;

Signed-off-by: Peng Ma <peng.ma@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 arch/arm/boot/dts/ls1021a.dtsi                 |  8 ++++++--
 arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi | 10 ++++++++--
 arch/arm64/boot/dts/freescale/fsl-ls1046a.dtsi | 10 ++++++++--
 3 files changed, 22 insertions(+), 6 deletions(-)

diff --git a/arch/arm/boot/dts/ls1021a.dtsi b/arch/arm/boot/dts/ls1021a.dtsi
index 13c768365680..b751a426e9ce 100644
--- a/arch/arm/boot/dts/ls1021a.dtsi
+++ b/arch/arm/boot/dts/ls1021a.dtsi
@@ -606,9 +606,13 @@
 			      <0x0 0x8389000 0x0 0x1000>, /* Status regs */
 			      <0x0 0x838a000 0x0 0x2000>; /* Block regs */
 			interrupts = <GIC_SPI 185 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 76 IRQ_TYPE_LEVEL_HIGH>;
-			interrupt-names = "qdma-error", "qdma-queue";
+				     <GIC_SPI 76 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 77 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "qdma-error",
+				"qdma-queue0", "qdma-queue1";
 			channels = <8>;
+			block-number = <2>;
+			block-offset = <0x1000>;
 			queues = <2>;
 			status-sizes = <64>;
 			queue-sizes = <64 64>;
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi
index 8d36f5044e93..a89c8a01d0f1 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi
@@ -728,9 +728,15 @@
 			      <0x0 0x8390000 0x0 0x10000>, /* Status regs */
 			      <0x0 0x83a0000 0x0 0x40000>; /* Block regs */
 			interrupts = <0 152 0x4>,
-				     <0 39 0x4>;
-			interrupt-names = "qdma-error", "qdma-queue";
+				     <0 39 0x4>,
+				     <0 40 0x4>,
+				     <0 41 0x4>,
+				     <0 42 0x4>;
+			interrupt-names = "qdma-error", "qdma-queue0",
+				"qdma-queue1", "qdma-queue2", "qdma-queue3";
 			channels = <8>;
+			block-number = <1>;
+			block-offset = <0x10000>;
 			queues = <2>;
 			status-sizes = <64>;
 			queue-sizes = <64 64>;
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1046a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1046a.dtsi
index 8e575dc57f02..b564d3fedad6 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1046a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1046a.dtsi
@@ -621,9 +621,15 @@
 			      <0x0 0x8390000 0x0 0x10000>, /* Status regs */
 			      <0x0 0x83a0000 0x0 0x40000>; /* Block regs */
 			interrupts = <0 153 0x4>,
-				     <0 39 0x4>;
-			interrupt-names = "qdma-error", "qdma-queue";
+				     <0 39 0x4>,
+				     <0 40 0x4>,
+				     <0 41 0x4>,
+				     <0 42 0x4>;
+			interrupt-names = "qdma-error", "qdma-queue0",
+				"qdma-queue1", "qdma-queue2", "qdma-queue3";
 			channels = <8>;
+			block-number = <1>;
+			block-offset = <0x10000>;
 			queues = <2>;
 			status-sizes = <64>;
 			queue-sizes = <64 64>;
-- 
2.17.0

